.github/workflows/macos-generation.yml

```diff
@@ -0,0 +1,198 @@
+name: macOS image generation
+on:
+  workflow_call:
+    inputs:
+      image_label:
+        type: string
+        description: macOS codename
+        required: true
+      base_image_name:
+        type: string
+        description: Base clean image
+        required: true
+      virtualMachineName:
+        type: string
+        description: Name for VM in vSphere
+        required: true
+      template_path:
+        type: string
+        description: Packer template path
+        required: true
+      target_datastore:
+        type: string
+        description: Image datastore
+        required: true
+      custom_repo:
+        type: string
+        description: Custom repo to checkout
+        required: false
+      custom_repo_hash:
+        type: string
+        description: Custom repo hash
+        required: false
+
+env:
+  KEYVAULT: imagegeneration
+  ESXI_CLUSTER: mcv2-build-unstable
+  VCENTER_DATACENTER: imagegen
+  OUTPUT_FOLDER: mms-output
+  BUILD_DATASTORE: ds-image
+
+defaults:
+  run:
+    shell: pwsh
+
+jobs:
+  build:
+    runs-on: maccloudv2
+    timeout-minutes: 1200
+    steps:
+    #- uses: actions/checkout@v3
+    - name: Checkout repository
+      if: ${{ (inputs.custom_repo == '' && inputs.custom_repo_hash == '') || github.event_name != 'pull_request_target'  }}
```

Comment:
Are you sure that this condition is correct?
I see a couple of cases where it won't work properly:
- `workflow_dispatch` event: in this case both `checkout repository` and `Checkout custom repository` will be invoked.
- `pull_request_target` event - the same.

May be move the calculation to the separate step?
```yml
- name: Determine checkout type
  run: |
    if ("${{ inputs.custom_repo }}" -and "${{ inputs.custom_repo_hash }}") {
        $checkoutType = "custom_repo"
    } elseif (("${{ github.event_name }}" -eq "pull_request_target") -and (${{ github.event.action }} -eq "labeled" )) {
        $checkoutType = "pull_request"
    } else {
        $checkoutType = "main"
    }
    "CHECKOUT_TYPE=$checkoutType" | Out-File -Append $env:GITHUB_ENV
```
And then use this variable in `if` conditions for checkout steps?
Right now, you have 3 separate conditions and it looks very unclear. Also, if you change one of them, you have to change two other conditions too.

Reply:
Updated according to suggestion